modules/cudf/src/series.ts

```diff
@@ -922,45 +961,56 @@ function inferType(value: any[]): DataType {
   let stringsCount  = 0;
   let bigintsCount  = 0;
   let booleansCount = 0;
-  let unknownCount  = 0;
+  let datesCount    = 0;
   value.forEach((val) => {
     if (val == null) { return ++nullsCount; }
     switch (typeof val) {
       case 'bigint': return ++bigintsCount;
       case 'boolean': return ++booleansCount;
       case 'number': return ++numbersCount;
       case 'string': return ++stringsCount;
-      case 'object': return Array.isArray(val) ? ++arraysCount : ++objectsCount;
+      case 'object':
+        if (Array.isArray(val)) {
+          ++arraysCount;
+        } else if (Object.prototype.toString.call(val) === '[object Date]') {
+          ++datesCount;
+        } else {
+          ++objectsCount;
+        }
+        return;
```

Comment:
This should fall through if not one of our recognized types, right?

```suggestion
      case 'object':
        if (Array.isArray(val)) {
          return ++arraysCount;
        } else if (Object.prototype.toString.call(val) === '[object Date]') {
          return ++datesCount;
        } else {
          return ++objectsCount;
        }
```